ansible_collections/arista/avd/plugins/plugin_utils/topology_generator_utils.py

```diff
@@ -0,0 +1,715 @@
+
+    Args:
+        filename: Name of the file
+
+    Returns:
+        Data from the file
+    """
+    with open(filename, "r", encoding="utf-8") as stream:
+        data = None
+        try:
+            data = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+        return data
+
+
+def create_node_dict(file_data, filename):
+    """
+    Create a dictionary of nodes with its corresponding neighbors
+
+    Args:
+        file_data: Data of structure config file
+        filename: Name of the file
+
+    Returns:
+        Dictionary with node and its corresponding neighbors
+    """
+    node_dict = {}
+    node_dict["edge"] = False
+    node_dict["name"] = os.path.splitext(filename)[0].split("/")[-1]
+    node_dict["neighbors"] = []
+    if file_data and file_data.get("ethernet_interfaces"):
+        if type(file_data["ethernet_interfaces"]) == dict:
+            for key, value in file_data["ethernet_interfaces"].items():
+                if "peer_interface" not in value:
+                    continue
+                neighbor_dict = {}
+                neighbor_dict["neighborDevice"] = value["peer"]
+                neighbor_dict["neighborPort"] = value["peer_interface"]
+                neighbor_dict["port"] = key
+                if "channel_group" in value and "id" in value["channel_group"]:
+                    neighbor_dict["portChannel"] = value["channel_group"]["id"]
+                node_dict["neighbors"].append(neighbor_dict)
+                if value["peer_type"] in EDGETYPES:
+                    node_dict["edge"] = True
+        else:
+            for value in file_data["ethernet_interfaces"]:
+                if "peer_interface" not in value:
+                    continue
+                neighbor_dict = {}
+                neighbor_dict["neighborDevice"] = value["peer"]
+                neighbor_dict["neighborPort"] = value["peer_interface"]
+                neighbor_dict["port"] = value["name"]
+                if "channel_group" in value and "id" in value["channel_group"]:
+                    neighbor_dict["portChannel"] = value["channel_group"]["id"]
+                node_dict["neighbors"].append(neighbor_dict)
+                if value["peer_type"] in EDGETYPES:
+                    node_dict["edge"] = True                
+
+    return node_dict
+
+
+def structured_config_to_topology_input(ol, node_dict, diagram_groups, current_dict=None):
+    """
+    Convert structure config file data to topology input data by creating a nested dictionary
+    of "nodes" and "groups"
+
+    Args:
+        ol: Original input dictionary with nested "name", "nodes" and "groups" keys
+        node_dict: List of nodes
+        node_neighbor_dict: Dictionary with nodes and neighbors list
+                       ex- {'0': ['super-spine1', 'super-spine2']}
+    Returns:
+       ol: List of dictionary with nested "name", "nodes" and "groups" keys
```

Comment:
is it the same as the input `ol`? if so indicates that the method modifies in place